### DISCUSSION TITLE
How to specify remote storage for target genomes in samplesheet?

### DISCUSSION THREAD

**Discussion by carbocation** [PGScatalog/pgsc_calc#290] [Q&A] [answered]
The [samplesheet setup](https://pgsc-calc.readthedocs.io/en/latest/how-to/samplesheet.html#setup-samplesheet) page states:
 
> ***path_prefix** should be set to the path of the target genomes excluding all file extensions*
>
> *Always use absolute paths that begin with /*

But if the target genomes are at a remote path (e.g., gs://bucket/path/target.pgen, gs://bucket/path/target.pvar, gs://bucket/path/target.psam ), how should this advice be operationalized? 

When I simply try to use "gs://bucket/path/target" as the prefix in the scorefile, it seems to be treated as a subfolder of the working directory. For example, since I am running from `/tmp/nextflow`, it becomes: `/tmp/nextflow/gs:/` in the error message.

In contrast, pointing to a score file in a gs:// bucket works just fine, so I don't think this is a fundamental limitation of nextflow.

(I considered mounting to make these look like local files with gcsfuse, but doing so is prohibited due to restricted permissions. And the pipeline is not being executed locally, so localizing these [enormous] genotype files to the launch machine is not sensible anyways.)

**Comment by smlmbrt** [comment]
Hi @carbocation, we have an open issue (https://github.com/PGScatalog/pgsc_calc/issues/288) to document how to use the pipeline with remote paths like that, it basically involves supplying a samplesheet in json format. @nebfield can explain more because he's the expert on getting it running in the cloud!

**Comment by nebfield** [comment, marked as answer]
You're right, nextflow natively supports object storage and cloud execution 🚀 

In `pgsc_calc` CSV samplesheets are incompatible with cloud storage, but you can use [JSON samplesheets](https://pgsc-calc.readthedocs.io/en/dev/how-to/cloud.html#your-genomes-are-in-cloud-storage) instead
